contracts/utils/InterchainTokenDeployer.sol

```diff
@@ -43,12 +42,18 @@ contract InterchainTokenDeployer is IInterchainTokenDeployer, Create3 {
         string calldata symbol,
         uint8 decimals
     ) external payable returns (address tokenAddress) {
-        bytes memory params = abi.encode(tokenManager, distributor, name, symbol, decimals);
         // slither-disable-next-line too-many-digits
-        bytes memory bytecode = bytes.concat(type(InterchainTokenProxy).creationCode, abi.encode(implementationAddress, params));
-
+        bytes memory bytecode = new bytes(0x37); //bytes.concat(type(InterchainTokenProxy).creationCode, abi.encode(implementationAddress, params));
+        address implementation = implementationAddress;
+        assembly {
+            mstore(add(bytecode, 0x20), shl(0x60, 0x3d602d80600a3d3981f3363d3d373d3d3d363d73))
```

Comment:
can you create constants for the magic numbers here?

Reply:
i don't see the point but we could